### DISCUSSION TITLE
System specifications required for development using remote ssh

### DISCUSSION THREAD

**Discussion by birdryoo** [microsoft/vscode-discussions#1164] [Extension Development QnA] [answered]
I use vscode on a Windows machine, and I enjoy working with the remote extension on a remote AWS ec2 machine.
However, the ec2 used for development mainly uses Amazon Linux with the lowest specifications, such as t2.nano or t2.micro, and this machine often stops as the CPU usage rate increases rapidly.
So, if I change the specifications to t2.small or t3.small that support 2G memory, the stopping phenomenon disappears.
This is a phenomenon that did not exist in the past and has been occurring for several months.

I wonder if it is possible to optimize or modify something so that remote extension can be used even with low server specifications.
It is true that the t2.small is cheaper, but it is also twice as expensive as the t2.micro.

**Comment by alefragnani** [comment, marked as answer]
Hi @birdryoo ,

Based on the [Remote Development using SSH / System Requirements](https://code.visualstudio.com/docs/remote/ssh#_system-requirements) documentation, the minimum requirements are _1 GB RAM is required for remote hosts, but at least 2 GB RAM and a 2-core CPU is recommended._ , And since you are referring to the **Remote - SSH** extension usage, I would suggest you to take a look on their repo (https://github.com/microsoft/vscode-remote-release/issues), which should have more information about it's usage. I did a quick search using terms like `cpu`, `minimum` and `memory` and a few issues appeared: 

* https://github.com/microsoft/vscode-remote-release/issues/2692
* https://github.com/microsoft/vscode-remote-release/issues/3298
* https://github.com/microsoft/vscode-remote-release/issues/9412

By the way, for future issues or feature requests about VS Code usage, please use the VS Code repository instead (https://github.com/microsoft/vscode/issues/new/choose). This Discussions repo is focused for extension development for now.

Hope this helps
